src/data/mv_polynomial/equiv.lean

```diff
@@ -144,6 +144,24 @@ def ring_equiv_congr [comm_semiring S₂] (e : R ≃+* S₂) :
   map_mul'  := ring_hom.map_mul _,
   map_add'  := ring_hom.map_add _ }
 
+/-- If `A` and `B` are isomorphic `R`-algebras, then we get an isomorphism
+`mv_polynomial S₁ A ≃ₐ[R] mv_polynomial B S₂`. -/
+def alg_equiv_congr {A B : Type*} [comm_semiring A] [comm_semiring B] [algebra R A] [algebra R B]
+  (e : A ≃ₐ[R] B) :
+  algebra.comap R A (mv_polynomial S₁ A) ≃ₐ[R] algebra.comap R B (mv_polynomial S₁ B) :=
```

Comment:
The docstring doesn't match the type. Do you also want an equiv `S1 =~ S2`?
(You also swapped `B` and `S2` in the docstring on the rhs.

Reply:
Fixed this, thank you! For the equivalence of the variables we already have `mv_polynomial.alg_equiv_of_equiv`, so I think we don't need to add it explicitly, but I can of course do it if you think it's better.

Reply:
Would it make sense to generalise that one, instead of adding this one? So that we have one version that can be used everywhere?

Reply:
This makes sense, but I think that most of the time people will want the isomorphism `mv_polynomial S₁ R ≃ₐ[R] mv_polynomial S₂ R` (using `R` as `R`-algebra on both sides). In practice this means having to write something like `alg_equiv_congr R alg_equiv.refl e`.

Reply:
So, how about `congr` is the general one, and then we have `congr_left` and `congr_right` as special cases. Then we still only have to prove the lemmas once.

Reply:
Now there is the general `alg_equiv_congr` and the two particular cases `alg_equiv_congr_left` and `alg_equiv_congr_right`.

I can do the same for rings (but the lemmas for rings are used in several files, so I will have to modify them too if I change the names).

Reply:
Did you want to make the same change for rings here, or separately? If you don't want to do it here, I'll delegate the merge. Otherwise, just mark for review again when that's done.

Reply:
I will open another PR, I am a little bit confused by the `@[simps]` for a `def` and I have to understand it better. It creates automatically the `apply` lemmas, right?